### PR TITLE
[Possibly Ready] Reworks the ebow to shoot faster and hit softer

### PR DESCRIPTION
**Pull request by anconfuzedrock**
## About The Pull Request
Ebow stores up to ten shots very similarly to the captain's laser now, except it's the energy crossbow. Projectiles do 13 toxin, 25 stamina, and don't have many of the disorienting effects it previously did. You can only buy one per uplink, and it does pathetic toxin damage on melee hit just for fluff.
## Why It's Good For The Game
I said I was going to change the ebow after the stun change, and, well, I did. Ebow's always been a very strong weapon thanks to its unlimited ammo supply an, well, the ebow dealing 65 damage total counting stam, giving every disorienting status, being silent, and causing vomiting was probably a bit too much to be reasonable to survive if you get hit with one bolt, especially since you won't dodge the next shot out of infinity- and since stuns were changed because one hit and you're done is unfair, I think "2 hits and you're basically screwed on dodging the second hit" was kind of dumb. Infinite ammo is enough of an edge over other traitor weapons. I always liked the ebow as a middleground among lethal and non lethal, highlighting how violent syndies really are. This should accentuate this (though if you're just trying to get someone down, this will deal just 50%  more damage in doing so as the current ebow does, and you probably won't have to shoot them again to keep due to the strength of stamina damage if you need them alive). This will give a reasonable chance to dodge or fight back, dealing less stamina damage than a disabler but being far quieter. What I think is the most important change, though, is that this makes the ebow less spammable- though it still recharges over time, you have motivation to hold your fire- if you run out, waiting for it to recharge mid combat is an awful idea. It charges fast enough for most planned assasinations, but if you're running out of maint and picking off one person a minute you're bound to run into issues, unless of course you have very good aim. This should dissuade ebow murderbone while keeping it a very solid pick for anyone with more nuanced plans.

## Changelog
:cl:
balance: the energy crossbow now charges up ten shots at once like it used to four years ago. Each shot does 25 stamina and 13 toxin, but recharges over time and can be fired more rapidly. 
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
TC cost increase seems like a strange thing to add on to these changes but otherwise this sounds pretty neat

**Comment by anconfuzedrock** [comment]
I mostly don't think it'd be very fun for some midget to run outta maint dual wielding ebows (WHICH WAS POSSIBLE WITH THE NERFED COST OF 10TC, AND EFFECTIVELY ALLOWED YOU TO STUN WITH THEM) every single round.

**Comment by Shadowflame909** [comment]
tbh I wish someone would just remove the e-bows lethal effects and gave it mute.


Like if Im buying an e-bow I dont want to murder this dude, I'm probably going to brainwash him though.

**Comment by 4dplanner** [comment]
> if Im buying an e-bow I dont want to murder this dude
lol

**Comment by anconfuzedrock** [comment]
I've kidnapped enough dozen people to know you can kidnap someone without murdering them, even if you don't pack charcoal. if you NEED to do no damage, there's the riot dart pistol (I plan on improving that next) I think that's the bit of theming and charm the game has with syndies, they don't make weapons to leave someone alive very often. The ebow's good for it, but not great. And with this change I think the ebow could be used both to kill and to kidnap.

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -545,11 +545,10 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 	It will synthesize and fire bolts tipped with a debilitating \
-	toxin that will damage and disorient targets, causing them to \
-	slur as if inebriated. It can produce an infinite number \
-	of bolts, but takes time to automatically recharge after each shot."
-	item = /obj/item/gun/energy/kinetic_accelerator/crossbow
-	cost = 10
+	toxin that will damage and weaken targets; It can store up to ten \
+	bolts, and recharges them over time."
+	item = /obj/item/gun/energy/crossbow
+	cost = 11
```
Comment (ShizCalev) [inline code/modules/uplink/uplink_items.dm]: changelog this too please.

Comment (anconfuzedrock) [reply code/modules/uplink/uplink_items.dm]: ye ye

**Comment by wesoda25** [comment]
11 is a bad number please make it 12

**Comment by anconfuzedrock** [comment]
Sorry wesoda, I feel sympathy for 11 and don't want it to be discarded so easily when it's clearly the perfect price.

**Comment by wesoda25** [comment]
If i see some loser in the uplink costing an odd 11 TC forcing to buy something worth 1 TC to even it out I will refuse to buy him make it 12!!!!

**Comment by anconfuzedrock** [comment]
the problem of nothing costing odd numbers will only be fixed if we start giving things prices based off their effects not off what's even, if that matters to people then maybe we should go back to a 10tc uplink.

**Comment by PKPenguin321** [comment]
>it's another "but you can buy 2!" episode

when the heck are we going to add a stock system for uplinks so you can only buy 1 of certain items without them costing >10 TCs

**Comment by Shadowflame909** [comment]
Based.


My first PR when that stock system gets added is making syndibombs only having 1 stock no matter the price.

Traitorlings becomes worse then terrible with syndibombs

**Comment by subject217** [comment]
@PKPenguin321 Great idea, when you code it

**Comment by anconfuzedrock** [comment]
what do these failed checks meaaan

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -79,35 +79,37 @@
...
 
-/obj/item/gun/energy/kinetic_accelerator/crossbow
+/obj/item/gun/energy/crossbow
 	name = "mini energy crossbow"
 	desc = "A weapon favored by syndicate stealth specialists."
 	icon_state = "crossbow"
 	item_state = "crossbow"
+	force = 4
+	damtype = TOX
+	attack_verb = list(""poisoned", "jabbed")
```
Comment (4dplanner) [inline code/modules/projectiles/guns/energy/special.dm]: ""Poisoned"

Comment (Mickyan) [inline code/modules/projectiles/guns/energy/special.dm]: extra quotation mark here

Comment (anconfuzedrock) [reply code/modules/projectiles/guns/energy/special.dm]: ye so

Comment (4dplanner) [reply code/modules/projectiles/guns/energy/special.dm]: The point is the two quotation marks

**Comment by PKPenguin321** [comment]
Wait what the fuck [we already have a limited stock system in the uplink code](https://github.com/tgstation/tgstation/blob/master/code/modules/uplink/uplink_items.dm#L71)

File: code/modules/uplink/uplink_items.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/uplink/uplink_items.dm]: Undo this change, on the next line insert:
`limited_stock = 1`

File: code/modules/projectiles/guns/energy/special.dm (hunk 2 above)
Comment (anconfuzedrock) [reply code/modules/projectiles/guns/energy/special.dm]: based and 4dpilled

File: code/modules/uplink/uplink_items.dm (hunk 1 above)
Comment (anconfuzedrock) [reply code/modules/uplink/uplink_items.dm]: But pkpenguiin I also like the price being 11tc regardless of whether or not you can buy two

Comment (PKPenguin321) [reply code/modules/uplink/uplink_items.dm]: I do not

Comment (anconfuzedrock) [reply code/modules/uplink/uplink_items.dm]: ebow has always been good as fuck mate I doubt this'll make it anything but less obvious a choice

Comment (PKPenguin321) [reply code/modules/uplink/uplink_items.dm]: RIP ebow i guess

**Comment by anconfuzedrock** [comment]
why is travis failin man it's freaking me ouuuutttttt

**Comment by necromanceranne** [comment]
TBH Rock, these changes seem entirely too soon after the previous changes to be justified. I feel like it would have been better to let the new version be tried out for a little while longer before seeing how things lie.

Also, the weapon is an assassination tool for holding them in place. It's core purpose is assassination. I don't really get how two shots is any less of a drawback than the previous version which was a hard stun akin to a taser. This version just doesn't really do the whole killing thing all that well either by making it fuck awful at disabling and fuck awful at killing. You're trying to give it a bit of both but it just sucks at both. And I'm not sure why you would take this over just blasting the boy with a gat. AP rounds in a stechkin blow people away pretty fast and pretty effectively. When I asked about the changes you said it would still disable. I didn't realize what you meant was 'it's shittier than even a disabler in every way and the disable bit is the stamina damage'. So that's...great....

In addition to this, the current version of the weapon is actually brilliant for kidnapping. It holds them in place for a good while once both shots go down, enough to cuff and remove headset, and even if you get just the first shot off you can even follow with a emp flashlight with relative ease if you wanna make sure they can't keep screaming for help.

This just makes the ebow a really fuck awful disabler gun and none of it's previous identity at all Casually plinking at the murder target is not something that justifies the 11tc cost, and the fact that you're charging more for it is...interesting. I'd say the current one is underpriced. This version is horrifically overpriced. I might even prefer a donksoft pistol if I was to be completely honest with you. At least then I can shoot fast enough and add a silencer to maybe kill a guy, and not run into problems energy weapons have like reflector vests.

**Comment by necromanceranne** [comment]
Actually no that's a good point I wanna highlight that.

A toy pistol does more stamina damage, fires as fast as a stechkin, can be silenced and is only 3 TC.

You could get the same results from a toy pistol. At a 75% discount no less. Rock, what the fuck are you doing man?

**Comment by anconfuzedrock** [comment]
I'm not charging more for it any more. You've never seen a donksoft pistol if you think it'd be better. Ebow will always have plenty of draws. Why am I talking like this. Maybe I should slightly increase that tox though.

**Comment by anconfuzedrock** [comment]
Literally try that fucking thing, though. the difference is you dump 4 shots outta eight on that into someone and they might keep standing. riot pistol is fucking garbage and another thing I'm thinking about.

**Comment by necromanceranne** [comment]
Then is it a murder weapon in of itself? Again, why am I not just using a stechkin? For the same reason I could use a toy pistol, the fuckers fire fast and can be silenced. If I truly wanted to kidnap, I'd be better served asking cargo for a disabler. It's not like they'd turn down emagging their console.

If I'm taking a silenced weapon I'm not looking to enter a firefight with someone I'm looking to make whatever I'm doing fast. This has none of these qualities and if you upped the toxins damage you're looking at a kinda shitty stechkin. We have the stechkin. We don't need another one. And these values seem quite similar to the initial ones actionninja floated in his own PR that got changed, minus any of the debilitating effects it throws on you. God forbid you encounter someone with an actual fucking weapon.

Also who buys two? Why keep it capped at one with this change?

**Comment by anconfuzedrock** [comment]
I think that infinite ammo for 2 disablers would be excruciatingly frustrating to people, Then again, that would be 20tc. I guess it was mostly me thinking about it from the actioninja version-being nutted on with 2 50stam shots would defeat the point of there not being stuns. I might change it. I'd rather this make the ebow a bit weak than too strong, one is far easier to fix. Nobody's seen this in action at all, and the main goals here are to make ebows promote murderbone and spam less while making it more reasonable to survive being shot with it, since that was the point behind the taser removal and thus previous ebow change.

**Comment by MrDoomBringer** [comment]
image

**Comment by HomocidlePancakes** [comment]
>Promote murderbone
making it more reasonable to survive being shot with it

Thats a self-contradiction right here

**Comment by anconfuzedrock** [comment]
I meant to not promote murderbone

**Comment by dr-bee** [comment]
I mean, 10 shots of 40 armor bypassing damage is pretty fucking good. Its enough to drop 3 people if you hit every shot.  It may not be as good as the stechkin in a stand up firefight but the stechkin is overtuned IMHO.

**Comment by HomocidlePancakes** [comment]
It's been less then a week since the last ebow rework. Too early to rework it again

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -79,35 +79,37 @@
...
 	name = "mini energy crossbow"
 	desc = "A weapon favored by syndicate stealth specialists."
 	icon_state = "crossbow"
 	item_state = "crossbow"
+	force = 4
+	damtype = TOX
+	attack_verb = list("poisoned", "jabbed")
 	w_class = WEIGHT_CLASS_SMALL
-	materials = list(MAT_METAL=2000)
+	cell_type = /obj/item/stock_parts/cell{charge = 1000; maxcharge = 1000}
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/special.dm]: I'm pretty sure there's a cell that already has these numbers. And again, this is gross. IF there's not, make a subtype and use it here, don't use curly brackets to set vars on a type.

Comment (anconfuzedrock) [reply code/modules/projectiles/guns/energy/special.dm]: my excuse is the mini-egun does it so I went off that.

**Comment by necromanceranne** [comment]
>I think that infinite ammo for 2 disablers would be excruciatingly frustrating to people,

As opposed to...what? Him shooting you at point blank with a revolver? Getting grabbed with sleeping carp? Tot equipment is meant to give them an edge over the crew when employed properly. That's why it's mostly balanced behind TC cost. Given rechargers are fuck-easy to get your hands on, two actual disablers and being a good shot in of itself is effectively infinite ammo. You can get this as a crew member and you can essentially kill everyone with that too. Why on earth should a disabler be worth 10 TC? Why not save the TC and buy one from cargo? Do you think anyone actually gives a fuck about cargo buying guns on /tg/?

If it's the concealable nature of the weapon I'd like to remind you that there is a guaranteed method of acquiring a concealable energy pistol from space with almost 100% certainty if you know what you're doing in extremely short order. I know of a few players who do it consistently. That one you can't even stop people from acquiring if they have ready access to EVA equipment or justification for using it, like the curator or engineering. Spending 8 TC on a hardsuit in fact even grants you a slightly slower jetpack but which only requires oxygen over Co2 to use, so getting an oxygen tank, which nobody in their right mind is going to say no to granting you, requires even less effort. This same ruins comes with a free esword. You've saved a lot of TC and at very minimal setup.

And if THAT isn't your ticket then a sawn-off improvised shotgun fired at point blank with rubber shot is actually straight up stamina crit. That can be hidden in fedoras and storage implants. It works just as well if you're taking someone by surprise. Using certain methods it's concealable to boot. Rubber shot only requires access to an autolathe, and not even a hacked autolathe.

**Comment by anconfuzedrock** [comment]
Ye I get it thus the me buffing it significantly from the start of the pr. This is still cheaper than the old ebow used to be. Week or not, the current design of ebow allowing you to instastun by dual wielding or to near instacrit with the large ebow shouldn't be around.

**Comment by necromanceranne** [comment]
You buy two ebows, you have no TC. You killing them with a toolbox while they can scream their current location the entire time?

And large Ebows you won't see until deep into most rounds. Illegal tech is a 10000 point node deep into the tech web. Nobody gets that shit quickly.

**Comment by anconfuzedrock** [comment]
Thus me making changes so it's possible to buy 2 ebows, though this version of ebows would certainly allow you to kill them with the gun itself instead of with a toolbox. I don't care what the fuck ebows are behind- illegal tech is easy enough for a syndie, and you can't possibly tell me that even deep into a round a single shot dealing 40 tox and 50 stamina is balanced (we removed laser cannons because 40 burn was a lot)

**Comment by Cyprex** [comment]
Use ```limited_stock = 1``` if you are worried about people buying two.

**Comment by anconfuzedrock** [comment]
I did, but changed my mind- the previous ebow could basically instastun with dial wielded bolts, but this one can’t.

**Comment by ExcessiveUseOfCobblestone** [comment]
Is the goal of this PR to rework the gun because you always thought it was bad or to tweak it out of response to the stun changes?

Personally if it's the latter, I'd prefer you make changes that are subtle enough to do the job without warranting of TC changes.

**Comment by Cyprex** [comment]
The TC cost isn't changed as of the latest state of the PR.

**Comment by anconfuzedrock** [comment]
A bit of both, really. Its current state is ridiculous, but its tc cost was changed to 10 by the stun rework pr. I’m not planning on touching it now.

**Comment by wesoda25** [comment]
Double ebow is gay by default and shouldn’t be allowed (short of discounts) with either form of the gun.

**Comment by necromanceranne** [comment]
I agree with wesoda, if the problem is two ebows then just make it 12 again. It honestly should be 12 anyway.

File: code/modules/projectiles/projectile/energy/ebow.dm [hunk 4]
```diff
@@ -4,14 +4,12 @@
...
+	stamina = 25
 
 /obj/item/projectile/energy/bolt/halloween
 	name = "candy corn"
 	icon_state = "candy_corn"
 
 /obj/item/projectile/energy/bolt/large
-	damage = 40
+	damage = 25
+	stamina = 20
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/projectile/energy/ebow.dm]: You'll be in crit faster than getting stamstunned, is that intended?

File: code/modules/projectiles/ammunition/energy/ebow.dm [hunk 5]
```diff
@@ -10,3 +10,4 @@
 /obj/item/ammo_casing/energy/bolt/large
 	projectile_type = /obj/item/projectile/energy/bolt/large
 	select_name = "heavy bolt"
+	e_cost = 100
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/ammunition/energy/ebow.dm]: you can remove this

File: code/modules/projectiles/projectile/energy/ebow.dm (hunk 4 above)
Comment (anconfuzedrock) [reply code/modules/projectiles/projectile/energy/ebow.dm]: I don’t think that’ll happen? Isn’t it the combined stam and damage? I do want the large ebow to be more lethal-the station doesn’t need more stuns- but maybe I’ll make it straight lethal or something.

Comment (PKPenguin321) [reply code/modules/projectiles/projectile/energy/ebow.dm]: Stam stun will happen first yes

Comment (ExcessiveUseOfCobblestone) [reply code/modules/projectiles/projectile/energy/ebow.dm]: oh right it's based on hp not maxhp

75 hp-20 stam
50-<40
**25-<60**

**Comment by anconfuzedrock** [comment]
CObbmommy’s word is law

**Comment by anconfuzedrock** [comment]
Right, I think this is everything, set tox to 13 instead of 15 so it isn't as good on the small bow, but this should be ready if I didn't forget anything else.

File: code/modules/projectiles/ammunition/energy/ebow.dm (hunk 5 above)
Comment (anconfuzedrock) [reply code/modules/projectiles/ammunition/energy/ebow.dm]: ye did

File: code/modules/projectiles/guns/energy/special.dm [hunk 6]
```diff
@@ -79,31 +79,31 @@
...
 	suppressed = TRUE
 	ammo_type = list(/obj/item/ammo_casing/energy/bolt)
 	weapon_weight = WEAPON_LIGHT
 	obj_flags = 0
-	overheat_time = 20
-	holds_charge = TRUE
-	unique_frequency = TRUE
 	can_flashlight = FALSE
-	max_mod_capacity = 0
+	selfcharge = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/guns/energy/special.dm]:
```suggestion
	selfcharge = TRUE
```

**Comment by ExcessiveUseOfCobblestone** [comment]
It works[tm] but this PR just makes me go "okay..." you feel

**Comment by anconfuzedrock** [comment]
Yeah, I’m the kind of person to be satisfied with “okay...”.

**Comment by ExcessiveUseOfCobblestone** [comment]
Do other people actually want this? I'm on the fence so it's time to phone some friends.

**Comment by ExcessiveUseOfCobblestone** [comment]
Doesn't seem that popular and I don't feel like it's warranting to overturn that.
